### PR TITLE
Support CSS animation for text-align-last and text-justify

### PR DESCRIPTION
**Pull request by tuankiet65**
#### 8fd683703b324056047dcbb939050b4b4f159eff
<pre>
Support CSS animation for text-align-last and text-justify
<a href="https://bugs.webkit.org/show_bug.cgi?id=240591">https://bugs.webkit.org/show_bug.cgi?id=240591</a>
&lt;rdar://93918638 &gt;

Patch by Kiet Ho &lt;tho22@apple.com &gt; on 2022-06-02
Reviewed by Tim Nguyen.

Tests: LayoutTests/imported/w3c/web-platform-tests/css/css-text/text-align/text-align-last-interpolation.html
       LayoutTests/imported/w3c/web-platform-tests/css/css-text/text-justify/text-justify-interpolation.html
       LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002.html
       LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002.html
       LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-text/text-align/text-align-last-interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/text-justify/text-justify-interpolation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:

Canonical link: <a href="https://commits.webkit.org/251235@main">https://commits.webkit.org/251235@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295146">https://svn.webkit.org/repository/webkit/trunk@295146</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Source/WebCore/animation/CSSPropertyAnimation.cpp [hunk 1]
```diff
@@ -3294,6 +3294,8 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
         new DiscretePropertyWrapper<RubyPosition>(CSSPropertyWebkitRubyPosition, &RenderStyle::rubyPosition, &RenderStyle::setRubyPosition),
         new DiscretePropertyWrapper<TableLayoutType>(CSSPropertyTableLayout, &RenderStyle::tableLayout, &RenderStyle::setTableLayout),
         new DiscretePropertyWrapper<TextAlignMode>(CSSPropertyTextAlign, &RenderStyle::textAlign, &RenderStyle::setTextAlign),
+        new DiscretePropertyWrapper<TextAlignLast>(CSSPropertyTextAlignLast, &RenderStyle::textAlignLast, &RenderStyle::setTextAlignLast),
+        new DiscretePropertyWrapper<TextJustify>(CSSPropertyTextJustify, &RenderStyle::textJustify, &RenderStyle::setTextJustify),
```
Comment (nt1m) [inline Source/WebCore/animation/CSSPropertyAnimation.cpp]: nit: place `text-justify` in alphabetical order

File: Source/WebCore/rendering/style/RenderStyleConstants.cpp [hunk 2]
```diff
@@ -1037,6 +1037,33 @@ TextStream& operator<<(TextStream& ts, TextDecorationStyle decorationStyle)
...
+    case TextAlignLast::Left: ts << "left"; break;
+    case TextAlignLast::Right: ts << "right"; break;
+    case TextAlignLast::Center: ts << "center"; break;
+    case TextAlignLast::Justify: ts << "justify"; break;
+    }
+
+    return ts;
+}
+
+TextStream& operator<<(TextStream& ts, TextJustify justify)
```
Comment (nt1m) [inline Source/WebCore/rendering/style/RenderStyleConstants.cpp]: ditto here, so after TextEmphasisPosition

**Review by nt1m — APPROVED** [review]
Perhaps @graouts wants to take another look.

**Comment by nt1m** [comment]
Looks like there are 3 tests you need to rebaseline (improvements) according to EWS:

imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002.html
imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002.html
imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002.html

**Comment by graouts** [comment]
@tuankiet65 I was out when this came up, but I wanted to thank you for fixing this. I love these kinds of fixes!

**Pull request merged**